cmd/hatchet-migrate/migrate/migrations/20250829173445_v1_0_41.sql

```diff
@@ -0,0 +1,526 @@
+-- +goose no transaction
+-- +goose Up
+-- +goose StatementBegin
+BEGIN;
+CREATE TABLE IF NOT EXISTS v1_lookup_table_partitioned (
+    tenant_id UUID NOT NULL,
+    external_id UUID NOT NULL,
+    task_id BIGINT,
+    dag_id BIGINT,
+    inserted_at TIMESTAMPTZ NOT NULL,
+
+    PRIMARY KEY (external_id, inserted_at)
+) PARTITION BY RANGE (inserted_at);
+
+SELECT create_v1_weekly_range_partition('v1_lookup_table_partitioned', NOW()::DATE);
+SELECT create_v1_weekly_range_partition('v1_lookup_table_partitioned', (NOW() - INTERVAL '1 week')::DATE);
+
+DO $$
+DECLARE
+    startDateStr varchar;
+    endDateStr varchar;
+    targetTableName CONSTANT varchar := 'v1_lookup_table_partitioned';
+    newTableName varchar;
+BEGIN
+    SELECT '19700101' INTO startDateStr;
+    SELECT TO_CHAR(date_trunc('week', (NOW() - INTERVAL '8 days')::DATE), 'YYYYMMDD') INTO endDateStr;
+    SELECT LOWER(FORMAT('%s_%s', targetTableName, startDateStr)) INTO newTableName;
+
+    EXECUTE
+        format('CREATE TABLE IF NOT EXISTS %s (LIKE %s INCLUDING INDEXES)', newTableName, targetTableName);
+
+    EXECUTE
+        format('ALTER TABLE %s SET (
+            autovacuum_vacuum_scale_factor = ''0.1'',
+            autovacuum_analyze_scale_factor=''0.05'',
+            autovacuum_vacuum_threshold=''25'',
+            autovacuum_analyze_threshold=''25'',
+            autovacuum_vacuum_cost_delay=''10'',
+            autovacuum_vacuum_cost_limit=''1000''
+        )', newTableName);
+    EXECUTE
+        format('ALTER TABLE %s ATTACH PARTITION %s FOR VALUES FROM (''%s'') TO (''%s'')', targetTableName, newTableName, startDateStr, endDateStr);
+END$$;
```

Comment:
Important: I don't know of a way to make this all `IF NOT EXISTS`-able, so we won't be able to rerun this safely AFAICT

Reply:
I think it's safe to check for whether we have the table already before `CREATE TABLE IF NOT EXISTS`, and if we do just return early? since these statements are all committed in the same tx